icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -658,12 +849,33 @@ class Parser {
             // continue;
         }
 
-        if (match < 0) {
-            status = kUnitIdentifierSyntaxError;
-        } else {
+        if (match >= 0) {
             fIndex = previ;
+            return {match};
         }
-        return {match};
+
+        // the index of the character after the last character of the constant denominator.
+        int32_t endOfConstantIndex = -1;
+        // If no match was found, we check if the token is a constant denominator.
+        // 1. find the first `-` from the `currentFIndex` to the end.
+        for (int32_t i = currentFIndex; i < fSource.length(); ++i) {
```

Comment:
Can't you use `sdrchr()` to do this?

Reply:
sure, I found that we can use `StringPiece::find` instead.